x/staking/src/keeper/unbonding.rs

```diff
@@ -83,56 +91,56 @@ impl<
         keys.iter().for_each(|k| {
             store.delete(k);
         });
-        Ok(mature_unbonds)
+        mature_unbonds
     }
 
     /// Insert an unbonding delegation to the appropriate timeslice in the unbonding queue
     pub fn insert_ubd_queue<DB: Database, CTX: TransactionalContext<DB, SK>>(
         &self,
         ctx: &mut CTX,
         delegation: &UnbondingDelegation,
-        time: chrono::DateTime<Utc>,
-    ) -> anyhow::Result<()> {
-        let time_slice = self.get_ubd_queue_time_slice(ctx, time);
+        time: Timestamp,
+    ) {
+        let time_slice = self.ubd_queue_time_slice(ctx, &time);
         let dv_pair = DvPair::new(
             delegation.validator_address.clone(),
             delegation.delegator_address.clone(),
         );
 
         if let Some(mut time_slice) = time_slice {
             time_slice.push(dv_pair);
-            self.set_ubd_queue_time_slice(ctx, time, time_slice)
+            self.set_ubd_queue_time_slice(ctx, time, time_slice);
         } else {
-            self.set_ubd_queue_time_slice(ctx, time, vec![dv_pair])
+            self.set_ubd_queue_time_slice(ctx, time, vec![dv_pair]);
         }
     }
 
     pub fn insert_unbonding_validator_queue<DB: Database, CTX: TransactionalContext<DB, SK>>(
         &self,
         ctx: &mut CTX,
         validator: &Validator,
-    ) -> anyhow::Result<()> {
-        let mut addrs = self.get_unbonding_validators(
-            ctx,
-            validator.unbonding_time,
-            validator.unbonding_height,
-        )?;
+    ) {
+        let mut addrs =
+            self.unbonding_validators(ctx, &validator.unbonding_time, validator.unbonding_height);
         addrs.push(validator.operator_address.to_string());
         self.set_unbonding_validators_queue(
             ctx,
-            validator.unbonding_time,
+            validator.unbonding_time.clone(),
             validator.unbonding_height,
             addrs,
-        )
+        );
     }
 
-    pub fn get_ubd_queue_time_slice<DB: Database, CTX: QueryableContext<DB, SK>>(
+    pub fn ubd_queue_time_slice<DB: Database, CTX: QueryableContext<DB, SK>>(
         &self,
         ctx: &mut CTX,
-        time: chrono::DateTime<Utc>,
+        time: &Timestamp,
     ) -> Option<Vec<DvPair>> {
         let store = ctx.kv_store(&self.store_key);
         let store = store.prefix_store(UBD_QUEUE_KEY);
+        let time = chrono::DateTime::from_timestamp(time.seconds, time.nanos as u32).expect(
+            "Expected correct conversion of timestamps. Invalid conversion can broke chain.",
```

Comment:
Can't the method accept a chrono::DateTime? We should try to use a single time type throughout the module rather than convert.

Reply:
I suggest that it is better to use domain or tendermint types in the public interfaces and make conversion for performing some operations. 
I can change the signature. Please, answer on the comment.